ci/test_conda_nightly_env.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+# Copyright (c) 2023, NVIDIA CORPORATION.
+
+set -euo pipefail
+
+RAPIDS_VERSION="23.12"
+CUDA_VERSION=${RAPIDS_CUDA_VERSION%.*}
+
+JSON_FILENAME="rapids_cuda${CUDA_VERSION}_py${RAPIDS_PY_VERSION}.json"
+
+echo "Creating conda environment with rapids=${RAPIDS_VERSION}, python=${RAPIDS_PY_VERSION}, cuda-version=${CUDA_VERSION}"
+#rapids-logger "Creating conda environment with rapids=${RAPIDS_VERSION}, python=${RAPIDS_PY_VERSION}, cuda-version=${CUDA_VERSION}"
+
+#rapids-conda-retry \
+conda \
+    create \
+    --solver=libmamba \
+    -n rapids-${RAPIDS_VERSION} \
+    -c rapidsai-nightly \
+    -c conda-forge \
+    -c nvidia  \
+    rapids=${RAPIDS_VERSION} \
```

Comment:
*(was going to put this as my review, realized a thread would probably be better)*

I totally support adding something like this, based on your description offline:

> _... to ensure that we are able to solve the full RAPIDS conda environment with recent packages (in other words, ensure there are no recent conflicts causing fallback to older conda packages)_

But I don't think it offers 100% protection against the case described in https://github.com/rapidsai/ops/issues/2947.

That issue is not about just packages building against too-old versions of dependencies... it's about packages across RAPIDS building against **very different**  versions of dependencies.

It looks to me that the code in this PR would catch cases like these:

- *"`rmm` nightlies haven't been published in the last 3 days"*
- *"the latest versions of `cugraph` and `pylibraft` can't be installed in the same environment"*

These aren't captured by the existing nightly tests at https://github.com/rapidsai/workflows/actions/workflows/nightly-pipeline.yaml.

But this wouldn't be guaranteed to catch a case like this:

- *"the latest `cuml` nightly built against an `rmm` from 5 days ago, but the latest `cudf` nightly built against an `rmm` from yesterday"*

Because this test with the `rapids` package is solving across all of the packages' **runtime** dependencies, but they could have ended up building against older versions based on conflicts in their individual **build** environments, right?

And those types of conflicts might not show up here if we use `pin_compatible(max_pin="x.x")` in `run` dependencies, e.g. `pylibraft==24.10.*` is going to have a runtime dependency on `rmm=24.10.*` regardless of which specific nightly of `rmm` it pulled in at build time. ([pylibraft nightly files](https://anaconda.org/rapidsai-nightly/pylibraft/files))

I think detecting that other case would have to happen at build time (or by post-processing of logs from build time). And I don't know how complex that would be, so can't say with confidence that the complexity would be worth it.

I totally support the approach this PR is pursuing, just wanted to be sure to note this other possible avenue for version mismatches to get through.

Reply:
You are correct about what this will and will not cover. I think it's worth pursuing because (1) it prevents runtime problems from being hidden and (2) sometimes runtime conflicts will also affect build environments, so this may give us a bit of signal into deeper problems should they arise.

Reply:
Ok great! I totally support moving forward with this.